src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,63 @@
 package practice;
 
-import java.util.Collections;
+import java.util.Arrays;
 import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
 import model.Candidate;
+import model.Cat;
 import model.Person;
 
 public class StreamPractice {
-    /**
-     * Given list of strings where each element contains 1+ numbers:
-     * input = {"5,30,100", "0,22,7", ...}
-     * return min integer value. One more thing - we're interested in even numbers.
-     * If there is no needed data throw RuntimeException with message
-     * "Can't get min value from list: < Here is our input 'numbers' >"
-     */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .flatMap(x -> Arrays.stream(x.split(",")))
+                .mapToInt(Integer::parseInt)
+                .filter(x -> x % 2 == 0)
+                .min()
+                .orElseThrow(() -> new RuntimeException("Can't get min value from list: "
+                        + "< Here is our input 'numbers' >"));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return numbers.stream()
+                .map(x -> numbers.indexOf(x) % 2 != 0 ? x - 1 : x)
+                .mapToInt(Integer::intValue)
+                .filter(x -> x % 2 != 0)
+                .distinct()
+                .average().orElseThrow(NoSuchElementException::new);
```

Comment:
```suggestion
                .map(n -> numbers.indexOf(n) % 2 != 0 ? n - 1 : n)
                .mapToInt(Integer::intValue)
                .filter(x -> x % 2 != 0)
                .average()
                .orElseThrow();
```
orElseThrow() throws NoSuchElementException by default